services/src/main/java/org/keycloak/services/resources/admin/RealmAdminResource.java

```diff
@@ -763,6 +763,35 @@ public Response testLDAPConnection(@QueryParam("action") String action, @QueryPa
         return result ? Response.noContent().build() : ErrorResponse.error("LDAP test error", Response.Status.BAD_REQUEST);
     }
 
+    /**
+     * Test SMTP connection with current logged in user
+     *
+     * @param settings SMTP settings
+     * @return
+     * @throws Exception
+     */
+    @Path("testSMTPConnection")
+    @GET
+    @NoCache
+    public Response testSMTPConnection(@QueryParam("settings") String settings) throws Exception {
```

Comment:
Check if password is equal to ComponentRepresentation.SECRET_VALUE. If it is then get the password from the realm model. That'll do the trick.